python/pyspark/sql/connect/dataframe.py

```diff
@@ -875,6 +875,30 @@ def to_jcols(
 
     melt = unpivot
 
+    def hint(self, name: str, *params: Any) -> "DataFrame":
+        """
+        Specifies some hint on the current DataFrame. As an example, the following code specifies
+        that one of the plan can be broadcasted: `df1.join(df2.hint("broadcast"))`
+
+        .. versionadded:: 3.4.0
+
+        Parameters
+        ----------
+        name: str
+            the name of the hint, for example, "broadcast", "SHUFFLE_MERGE" and "shuffle_hash".
+        params: tuple
+            the parameters of the hint
```

Comment:
I know that the documentation is most likeley directly from PySpark, but I'm wondering if we can add more context around what types can the params have? If I read through the code it can be `any` here but later only `Optional[Union[str, int]]`?

Reply:
Yeah, It's worth improving the docs of the parameters.